Connect api

samuelkinyanjui opened the pull request:
Connect Redux with an API

samuelkinyanjui committed:
Created slice

Created two slice
books
and categories
linked the slice to the store
Added the store to application

samuelkinyanjui committed:
Fixed errors

Fixed style eerors

samuelkinyanjui committed:
Did the initial setup

Setup a Redux store
Create a slice to save your state and reducers

samuelkinyanjui committed:
updated  to add components

Use Redux in React components.

samuelkinyanjui committed:
installed react dom

installed react dom

samuelkinyanjui committed:
Created Connect API

Connect Redux with an API

DammyShittu reviewed (commented):
Hi @samuelkinyanjui,

While you made a great effort in this project, I cannot proceed to review your code.

## Invalid Code Review Request

### Kindly fix your live demo or remove the section and header link on the readme.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

samuelkinyanjui committed:
updated the read me file

updated the read me file to remove the live Demo
Update other important Information on readme file

V-Blaze commented on src/redux/books/booksSlice.js:
```diff
@@ -0,0 +1,99 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+
```
- You have done great so far  :+1: :+1: But there are a few changes and requirements you need to meet before moving to the next milestone. :+1: 

- [ ] Looking at the video below, you can see that I get an error when trying to start the app. The first error points to an asset you are importing that doesn't exist. For the second error, you are using `axios` to make your API requests but you do not have axios installed in your project. I'll suggest you install axios from NPM to rectify this issue. :+1: :small_airplane:  

- - Video reference:

https://github.com/samuelkinyanjui/bookstore_add_reducers/assets/107632780/d6557e97-b921-474f-b631-3c9286e8cfb0

V-Blaze commented on src/redux/books/bookslice.js:
```diff
@@ -0,0 +1,40 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+// STATE
+const initialBooksState = [
+  {
+    item_id: 'item1',
+    title: 'The Great Gatsby',
+    author: 'John Smith',
+    category: 'Fiction',
+  },
+  {
+    item_id: 'item2',
+    title: 'Anna Karenina',
+    author: 'Leo Tolstoy',
```
- [ ] This was the initial code for your `bookSlice`, you were meant to continue building out your `bookslice` from this file and not create a new one entirely. But since you have created a new file, I'll suggest you delete this as it is of no use to the project anymore. :+1: :100: 

- [ ] ALso the reducer.js file in the redux folder appears to be empty and of no use. I'll suggest you delete it. :small_airplane: :wave:

V-Blaze commented on src/components/Book.js:
```diff
@@ -0,0 +1,42 @@
+// Individual Book component
+// import React from 'react';
```
- [ ] Leaving unused code as a comment in your codebase is not a best practice, I'll suggest you delete this. :+1: :+1:

V-Blaze reviewed (changes requested):
## Hi @samuelkinyanjui 

## Required changes 🖊️ 
# Almost there 🟡 🟦 🟨 

![giphy](https://user-images.githubusercontent.com/107632780/229314554-3700de13-2c00-46d5-bc77-46ad09af7467.gif)


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing the project requirements, but there are still some issues I think you need to improve on. 😄 👋 


 ## Highlights
- Readme looks professional ✔
-  Correct Git flow👋 
-  Meaningful branch name✔

 ## Required Changes ♻️

## Optional suggestions
- [OPTIONAL] I noticed that some of the commit messages are in the past tense (e.g., "Created slice ") rather than the present imperative tense. It's a good practice to use the present imperative tense for commit messages (e.g: 'Create bookstore slice') as it helps in making the commit messages more actionable and descriptive. :+1: :smile: 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

samuelkinyanjui committed:
Updated the appication

Installed Axios
Deleted Bookslice
Deleted reducer.js
Removed comments on code

V-Blaze commented on src/components/Nav.js:
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+// import userIcon from '../Assets/user-regular.svg';
+import logo from '../Assets/logo.svg';
```
- [ ] You have done great installing axios... But as I pointed out in the previous review, the error with the assets import while starting the app is still there.. I'll suggest you check the import path correctly. 👍 Also I'll suggest you actually run and test your project properly to make sure it works as expected before requesting for review.. 💯 😄

V-Blaze reviewed (changes requested):
## Hi @samuelkinyanjui 

# Almost there 🟡 

![1094a3e620ebea4cd6cf2d0aceaa8f52](https://user-images.githubusercontent.com/107632780/229645999-eb4dcb1d-5a0c-4bdd-a150-2ab5cee3f40d.gif)


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing some changes from the previous reviewers, but there are still some issues I think you need to improve on. 😄 👋 


 ## Highlights
- Readme looks professional ✔
-  Correct Git flow👋 
-  Meaningful branch name✔

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

samuelkinyanjui committed:
Updated the assets

updated the asset link

samuelkinyanjui committed:
Delete .eslintrc.json

removed one .eslintrc.json files as there was error for two files

Johnadibe reviewed (approved):
# Status: Approved ✔️
### Hi @samuelkinyanjui👋🏽, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

![LeonardoInGatsby](https://user-images.githubusercontent.com/38088818/233992044-ad00b20f-3f2f-4c86-8b94-1e613432af5f.gif)

## Highlights
- Implemented all changes requested by the previous reviewer✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
